scalikejdbc-streams/src/main/scala/scalikejdbc/streams/StreamingSQL.scala

```diff
@@ -0,0 +1,54 @@
+package scalikejdbc.streams
+
+import scalikejdbc.{ DBSession, SQL, WithExtractor, WrappedResultSet }
+
+trait StreamingSQL[A, E <: WithExtractor] {
+  private[streams] def underlying: SQL[A, E]
+
+  def statement: String
+
+  def parameters: Seq[Any]
+
+  def extractor: WrappedResultSet => A
+
+  protected[streams] def setSessionAttributes(session: DBSession): DBSession = {
+    session
+      .fetchSize(underlying.fetchSize)
+      .tags(underlying.tags: _*)
+      .queryTimeout(underlying.queryTimeout)
+  }
+}
+
+final class CursorStreamingSQL[A, E <: WithExtractor](s: SQL[A, E], fetchSize: Int) extends StreamingSQL[A, E] {
```

Comment:
There is no reason to put this class into this source file. Please move this to `CursorStreamingSQL.scala`.